modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/js/recurrence.js

```diff
@@ -485,6 +493,111 @@ AUI.add(
 						return datePicker;
 					},
 
+					_setDaysOfWeek: function(value) {
+						var instance = this;
+
+						var dayOfWeekNodes = instance.get('daysOfWeekCheckboxes').filter(':not([disabled])');
+
+						dayOfWeekNodes.each(
+							function(node, index) {
+								var check = value.indexOf(node.get('value')) > -1;
+
+								node.set('checked', check);
+							}
+						);
+
+						return value;
+					},
+
+					_setFrequency: function(value) {
+						var instance = this;
+
+						var frequencySelect = instance.get('frequencySelect');
+
+						frequencySelect.set('value', value);
```

Comment:
@brandizzi. I am working on a refactoring of the _onInputChange method. Maybe I will need your help to test it. OK?